lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -539,6 +539,9 @@ def pass_hook_checks(self, event):
             logger.info("skipping %s is only be executed by shards", type(event))
             return False
 
+        if not event.relation.app:
```

Comment:
When would this happen?

Reply:
Good question - I am embarrassed to say that I do not recall, but that I experienced and error related to this last week when testing by hand. If I had to guess it would be that unit_changed events occured resulting in a `relation_changed` event where the app is not present